docs/posts/backend-agnostic-arrays/index.qmd

```diff
@@ -0,0 +1,613 @@
+
+This is computed using the
+[`length`](../../reference/expression-collections.qmd#ibis.expr.types.arrays.ArrayValue.length)
+API on array expressions:
+
+::: {.panel-tabset}
+
+## BigQuery
+
+```{python}
+(
+    bq_ents.select("primary_name", num_titles=_.known_for_titles.length())
+    .order_by(_.num_titles.desc())
+    .limit(5)
+)
+```
+
+## DuckDB
+
+```{python}
+(
+    ddb_ents.select("primary_name", num_titles=_.known_for_titles.length())
+    .order_by(_.num_titles.desc())
+    .limit(5)
+)
+```
+
+:::
+
+It seems like the length of the `known_for_titles` might be capped at five!
+
+### Index
+
+We can see the position of `"actor"` or `"actress"` in `primary_profession`s:
+
+::: {.panel-tabset}
+
+## BigQuery
+
+```{python}
+bq_ents.primary_profession.index("actor")
+```
+
+```{python}
+bq_ents.primary_profession.index("actress")
+```
+
+## DuckDB
+
+```{python}
+ddb_ents.primary_profession.index("actor")
+```
+
+```{python}
+ddb_ents.primary_profession.index("actress")
+```
+
+:::
+
+A return value of `-1` indicates that `"actor"` is not present in the value:
+
+Let's look for entities that are not primarily actors:
+
+We can do this using the
+[`index`](../../reference/expression-collections.qmd#ibis.expr.types.arrays.ArrayValue.index)
+method by checking whether the positions of the strings `"actor"` or
+`"actress"` are both greater than 0:
+
+::: {.panel-tabset}
+
+## BigQuery
+
+```{python}
+actor_index = bq_ents.primary_profession.index("actor")
+actress_index = bq_ents.primary_profession.index("actress")
+
+bq_not_primarily_acting = (actor_index > 0) & (actress_index > 0)
+bq_not_primarily_acting.mean()
+```
+
+## DuckDB
+
+```{python}
+actor_index = ddb_ents.primary_profession.index("actor")
+actress_index = ddb_ents.primary_profession.index("actress")
+
+ddb_not_primarily_acting = (actor_index > 0) & (actress_index > 0)
+ddb_not_primarily_acting.mean()
+```
+
+:::
+
+Who are they?
+
+::: {.panel-tabset}
+
+## BigQuery
+
+```{python}
+bq_ents[bq_not_primarily_acting].order_by("nconst")
+```
+
+## DuckDB
+
+```{python}
+ddb_ents[ddb_not_primarily_acting].order_by("nconst")
+```
+
+:::
+
+It's not 100% clear whether the order of elements in `primary_profession` matters here.
+
+### Containment
+
+We can get people who are listed as actors or actresses using `contains`:
+
+::: {.panel-tabset}
+
+## BigQuery
+
+```{python}
+bq_non_actors = bq_ents[
+    ~_.primary_profession.contains("actor") & ~_.primary_profession.contains("actress")
+]
+bq_non_actors.order_by("nconst")
+```
+
+## DuckDB
+
+```{python}
+ddb_non_actors = bq_ents[
+    ~_.primary_profession.contains("actor") & ~_.primary_profession.contains("actress")
+]
+ddb_non_actors.order_by("nconst")
+```
+
+:::
+
+### Element removal
+
+We can remove elements from arrays too.
+
+::: {.callout-note}
+## [`remove()`](../../reference/expression-collections.qmd#ibis.expr.types.arrays.ArrayValue.remove) does not mutate the underlying data
```

Comment:
what does it do?

Reply:
Hm, I thought I showed an example of it. Guess not. I'd rather leave it as is though since that's what's in the original.